cached table: clarify usage scenarios (#16089)
pingcap/docs: cached-tables.md

The cached table feature is suitable for tables with the following characteristics:

> - The data volume of the table is small, for example, below 4 MiB.
> - The table is read-only or rarely updated, for example, with a write QPS (queries per second) of less than 10 times per minute.
> - The table is frequently accessed, and you expect a better read performance, for example, when facing hotspots on small tables while directly reading from TiKV.

Reviewer: ```suggestion
- The data volume of the table is small, for example, less than 4 MiB.
- The table is read-only or rarely updated, for example, with a write QPS (queries per second) of less than 10 times per minute.
- The table is frequently accessed, and you expect a better read performance, for example, when encountering hotspots on small tables during direct reads from from TiKV.
```
